core/modules/replica/DatabaseServicesMySQL.cc

```diff
@@ -2230,6 +2230,7 @@ void DatabaseServicesMySQL::saveIngestParam(string const& database,
                         conn->sqlEqual("database", database) + " AND " + 
                         conn->sqlEqual("category", category) + " AND " +
                         conn->sqlEqual("param", param);
+                conn->rollback();
```

Comment:
This bug fix was required because it's done in the following context:
```
        _conn->executeInsertOrUpdate(
            [&](decltype(_conn) conn) {
                conn->begin();
                conn->executeInsertQuery(...);
                conn->commit();
            },
            [&](decltype(_conn) conn) {
                conn->rollback();
                conn->begin();
                conn->executeSimpleUpdateQuery(...);
                conn->commit();
            }
        );
```
Method `executeInsertOrUpdate` requires a user to provide 2 `lambda` functions, the first one for attempting to insert something into a database, the second one - is a failover to update something which already exist in the database instead of inserting. When executing the first lambda, the operation `executeInsertOrUpdate` will catch the `DuplicateKey` exception, and if the one was thrown then it would invoke the second (`update`) lambda. Note that the method requires a user to do his/her own transaction management.That was a reason behind adding the `rollback` to cleanup anything which might be modified in the database by the `insert` lambda.  Otherwise `conn->begin()` in the `update` lambda would throw a "logic error" exception.

Note that `executeInsertOrUpdate` would also do automatic reconnects to MySQL should the connection be lost. This would also invalidate any transactions. That's the whole point in handing lambdas instead of single queries to that method.